src/bridge/transactions/take_2.rs

```diff
@@ -115,37 +135,46 @@ impl Take2Transaction {
             tx: Transaction {
                 version: bitcoin::transaction::Version(2),
                 lock_time: absolute::LockTime::ZERO,
-                input: vec![_input0, _input1, _input2],
-                output: vec![_output0],
+                input: vec![_input_0, _input_1, _input_2, _input_3],
+                output: vec![_output_0],
             },
             prev_outs: vec![
                 TxOut {
-                    value: input0.amount,
+                    value: input_0.amount,
                     script_pubkey: connector_0.generate_taproot_address().script_pubkey(),
                 },
                 TxOut {
-                    value: input1.amount,
-                    script_pubkey: connector_2.generate_address().script_pubkey(),
+                    value: input_1.amount,
+                    script_pubkey: connector_4.generate_address().script_pubkey(),
                 },
                 TxOut {
-                    value: input2.amount,
-                    script_pubkey: connector_3.generate_taproot_address().script_pubkey(),
+                    value: input_2.amount,
+                    script_pubkey: connector_5.generate_taproot_address().script_pubkey(),
+                },
+                TxOut {
+                    value: input_3.amount,
+                    script_pubkey: connector_c.generate_taproot_address().script_pubkey(),
                 },
             ],
             prev_scripts: vec![
-                connector_0.generate_taproot_leaf_script(0),
-                connector_2.generate_script(),
-                connector_3.generate_taproot_leaf_script(0),
+                connector_0.generate_taproot_leaf_script(input_0_leaf),
+                connector_4.generate_script(),
+                connector_5.generate_taproot_leaf_script(input_2_leaf),
+                connector_c.generate_taproot_leaf_script(input_3_leaf),
             ],
             connector_0,
-            connector_3,
+            connector_4,
+            connector_5,
+            connector_c,
             musig2_nonces: HashMap::new(),
             musig2_nonce_signatures: HashMap::new(),
             musig2_signatures: HashMap::new(),
         }
     }
 
-    fn sign_input0(&mut self, context: &VerifierContext, secret_nonce: &SecNonce) {
+    pub fn num_blocks_timelock_1(&self) -> u32 { self.connector_4.num_blocks_timelock }
```

Comment:
Pls double check that this fn is used anywhere.

Reply:
Line 903 graphs/peg_out